lib/exprotobuf.ex

```diff
@@ -46,9 +48,61 @@ defmodule Protobuf do
         end
     end
 
-    config |> parse(__CALLER__) |> Builder.define(config)
+    with {:ok, record_path} <- compile(:proto, opts[:from], opts[:other]),
+         [ok: erl_module]   <- compile(:bytecode, record_path),
+         config             <- %Config{config | erl_module: erl_module},
+      do: config |> parse(__CALLER__) |> Builder.define(config)
+  end
+
+  def compile(:proto, path, options) when is_binary(path) do
+    with {:ok, paths} <- extract_paths(path),
+         true         <- File.exists?(paths["fullpath"]) do
+      compile(:proto, paths, options)
+    else
+      :error -> {:error, "Something went wrong!"}
+    end
   end
 
+  def compile(:proto, paths, options) when is_map(paths) do
+    record_path = ~w(priv gpb_records)
+    |> Path.join
+    |> Path.expand(File.cwd!)
+    |> String.to_char_list
```

Comment:
`String.to_char_list` has been deprecated recently, should use `String.to_charlist` now